packages/nimble-components/src/rich-text/editor/tests/rich-text-editor.spec.ts

```diff
@@ -390,6 +390,79 @@ describe('RichTextEditor', () => {
             ]);
         });
 
+        describe('should render as a plain text for bold and italics input rule entered into the editor', () => {
+            const markdownInput: { name: string, input: string }[] = [
+                { name: 'bold(**)', input: '**bold**' },
+                { name: 'bold(__)', input: '__bold__' },
+                { name: 'italics(*)', input: '*italics*' },
+                { name: 'italics(_)', input: '_italics_' }
+            ];
+            const focused: string[] = [];
+            const disabled: string[] = [];
+
+            for (const value of markdownInput) {
+                const specType = getSpecTypeByNamedList(
+                    value,
+                    focused,
+                    disabled
+                );
+                specType(
+                    `for ${value.name} markdown input to the editor`,
+                    // eslint-disable-next-line @typescript-eslint/no-loop-func
+                    async () => {
+                        await pageObject.setEditorTextContent(value.input);
+
+                        expect(pageObject.getEditorTagNames()).toEqual(['P']);
+                        expect(pageObject.getEditorLeafContents()).toEqual([
+                            value.input
+                        ]);
+                    }
+                );
+            }
+        });
+
+        describe('should render as lists when its input rule is entered into the editor', () => {
+            const markdownInput: {
+                name: string,
+                input: string,
+                tagName: string
+            }[] = [
+                { name: 'bullet list', input: '*', tagName: 'UL' },
+                { name: 'bullet list', input: '+', tagName: 'UL' },
+                { name: 'bullet list', input: '-', tagName: 'UL' },
+                { name: 'numbered list', input: '1.', tagName: 'OL' },
+                { name: 'numbered list', input: '5.', tagName: 'OL' }
+            ];
+            const focused: string[] = [];
+            const disabled: string[] = [];
+
+            for (const value of markdownInput) {
+                const specType = getSpecTypeByNamedList(
+                    value,
+                    focused,
+                    disabled
+                );
+                specType(
+                    `for ${value.name} markdown input (${value.input}) to the editor`,
+                    // eslint-disable-next-line @typescript-eslint/no-loop-func
+                    async () => {
+                        await pageObject.setEditorTextContent(value.input);
+                        await pageObject.pressEnterKeyInEditor();
+                        await pageObject.setEditorTextContent(value.name);
+
+                        expect(pageObject.getEditorTagNames()).toEqual([
```

Comment:
I think this assertion will succeed for both of these DOM structures:
**Value is in the list**
```
<ol>
   <li><p>value</p></li>
</ol>
```

**Value is after the list**
```
<ol>
   <li></li>
</ol>
<p>value</p>
```

Should we use `getEditorTagNamesWithClosingTags()` for this test? Or remove the line that does ` await pageObject.setEditorTextContent(value.name);` and just assert that there is an empty list item?

Reply:
I have updated the test case with the `getEditorTagNamesWithClosingTags()` page object method.